src/vs/workbench/services/workingCopy/common/workingCopyFileService.ts

```diff
@@ -284,36 +310,38 @@ export class WorkingCopyFileService extends Disposable implements IWorkingCopyFi
 		return stat;
 	}
 
-	async move(files: Required<SourceTargetPair>[], options?: { overwrite?: boolean, undoRedoGroupId?: number, isUndoing?: boolean }, token?: CancellationToken): Promise<IFileStatWithMetadata[]> {
-		return this.doMoveOrCopy(files, true, options, token);
+	async move(operations: IMoveOperation[], token?: CancellationToken): Promise<IFileStatWithMetadata[]> {
+		return this.doMoveOrCopy(operations, true, token);
 	}
 
-	async copy(files: Required<SourceTargetPair>[], options?: { overwrite?: boolean, undoRedoGroupId?: number, isUndoing?: boolean }, token?: CancellationToken): Promise<IFileStatWithMetadata[]> {
-		return this.doMoveOrCopy(files, false, options, token);
+	async copy(operations: ICopyOperation[], token?: CancellationToken): Promise<IFileStatWithMetadata[]> {
+		return this.doMoveOrCopy(operations, false, token);
 	}
 
-	private async doMoveOrCopy(files: Required<SourceTargetPair>[], move: boolean, options?: { overwrite?: boolean, undoRedoGroupId?: number, isUndoing?: boolean }, token?: CancellationToken): Promise<IFileStatWithMetadata[]> {
-		const overwrite = options?.overwrite;
+	private async doMoveOrCopy(operations: IMoveOperation[] | ICopyOperation[], move: boolean, token?: CancellationToken): Promise<IFileStatWithMetadata[]> {
 		const stats: IFileStatWithMetadata[] = [];
 
 		// validate move/copy operation before starting
-		for (const { source, target } of files) {
-			const validateMoveOrCopy = await (move ? this.fileService.canMove(source, target, overwrite) : this.fileService.canCopy(source, target, overwrite));
+		for (const operation of operations) {
```

Comment:
Could benefit from some destructuring when you loop over operations? Try with `for (const { file: { source, target }, options } of operations) {` and you get every property you need in one go.